src/libraries/Common/src/System/Security/Cryptography/MLDsaImplementation.Windows.cs

```diff
@@ -2,47 +2,216 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
+using Internal.NativeCrypto;
+using Microsoft.Win32.SafeHandles;
+
+using NTSTATUS = Interop.BCrypt.NTSTATUS;
 
 namespace System.Security.Cryptography
 {
     internal sealed partial class MLDsaImplementation : MLDsa
     {
-        internal static partial bool SupportsAny() => false;
+        private static readonly SafeBCryptAlgorithmHandle? s_algHandle = OpenAlgorithmHandle();
+
+        private readonly bool _hasSeed;
+        private readonly bool _hasSecretKey;
+        private SafeBCryptKeyHandle _key;
 
-        // TODO: Define this in terms of Windows BCrypt.dll (ephemeral keys)
+        private MLDsaImplementation(
+            MLDsaAlgorithm algorithm,
+            SafeBCryptKeyHandle key,
+            bool hasSeed,
+            bool hasSecretKey)
+            : base(algorithm)
+        {
+            _key = key;
+            _hasSeed = hasSeed;
+            _hasSecretKey = hasSecretKey;
+        }
+
+        internal static partial bool SupportsAny() => s_algHandle is not null;
 
         protected override void SignDataCore(ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, Span<byte> destination) =>
-            throw new PlatformNotSupportedException();
+            Interop.BCrypt.BCryptSignHashPure(_key, data, context, destination);
 
         protected override bool VerifyDataCore(ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, ReadOnlySpan<byte> signature) =>
-            throw new PlatformNotSupportedException();
+            Interop.BCrypt.BCryptVerifySignaturePure(_key, data, context, signature);
+
+        internal static partial MLDsaImplementation GenerateKeyImpl(MLDsaAlgorithm algorithm)
+        {
+            Debug.Assert(s_algHandle is not null, $"Check {SupportsAny}() before calling.");
+
+            string parameterSet = PqcBlobHelpers.GetParameterSet(algorithm);
+            SafeBCryptKeyHandle keyHandle = Interop.BCrypt.BCryptGenerateKeyPair(s_algHandle);
+
+            try
+            {
+                Interop.BCrypt.BCryptSetSZProperty(keyHandle, Interop.BCrypt.BCryptPropertyStrings.BCRYPT_PARAMETER_SET_NAME, parameterSet);
+                Interop.BCrypt.BCryptFinalizeKeyPair(keyHandle);
+            }
+            catch
+            {
+                keyHandle?.Dispose();
+                throw;
+            }
+
+            return new MLDsaImplementation(algorithm, keyHandle, hasSeed: true, hasSecretKey: true);
+        }
+
+        internal static partial MLDsaImplementation ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            Debug.Assert(s_algHandle is not null, $"Check {SupportsAny}() before calling.");
```

Comment:
Did you mean to use nameof here?

```suggestion
            Debug.Assert(s_algHandle is not null, $"Check {nameof(SupportsAny)}() before calling.");
```